mono/metadata/metadata.c

```diff
@@ -2438,7 +2420,8 @@ delete_image_set (MonoImageSet *set)
 	for (i = 0; i < set->nimages; ++i)
 		set->images [i]->image_sets = g_slist_remove (set->images [i]->image_sets, set);
 
-	g_ptr_array_remove (image_sets, set);
+	if (image_sets)
```

Comment:
I was thinking that technically you could call this function out of order with an improperly created image set, but the function isn't visible outside this file and all uses in the file are correct so I guess that isn't a real concern.